### PR TITLE
Overall code performance and reliability greatly increased

### PR DESCRIPTION
**Pull request by ferib**
Thomas is wearing glasses,
Thomas wants to see sharp.

### CODE REVIEW

**Comment by meyskens** [comment]
Mist deployment configuratie en een CI update.

File: Attributes.cs [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+        public sealed class GuildOnly : CheckBaseAttribute
+        {
+            public override Task<bool> ExecuteCheckAsync(CommandContext ctx, bool help)
+                => Task.FromResult(ctx.Guild != null || help);
+        }
+
+        /// <summary>
+        /// Defines that usage of this command is restricted to the bot-channel. Developers can override this.
+        /// </summary>
+        [AttributeUsage(AttributeTargets.Method | AttributeTargets.Class, AllowMultiple = false, Inherited = true)]
```
Comment (meyskens) [inline Attributes.cs]: we do not restrict channel usage

File: Commands/AdminCommands.cs [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+namespace ThomasBot.Commands
+{
+
+    [RequirePermissions(Permissions.Administrator)]
+    class AdminCommands : BaseCommandModule
+    {
+        [Command("mute")]
+        [Description("Een gebruiker muten (admin only)")]
+        [RequireBotPermissions(Permissions.ManageRoles)]
+        public async Task Mute(CommandContext ctx, DiscordMember member)
```
Comment (meyskens) [inline Commands/AdminCommands.cs]: This mutes the wrong user

File: Commands/AdminCommands.cs [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+            await ctx.RespondAsync(":mute:");
+        }
+
+        [Command("unmute")]
+        [Description("Een gebruiker unmuten (admin only)")]
+        [RequireBotPermissions(Permissions.ManageRoles)]
+        public async Task Unmute(CommandContext ctx, DiscordMember member)
+        {
+            var role = ctx.Guild.Roles.First(x => x.Value.Name == "Muted").Value;
+            await member.GrantRoleAsync(role);
```
Comment (meyskens) [inline Commands/AdminCommands.cs]: Twice `GrantRole`

**Comment by meyskens** [comment]
Screenshot from 2020-03-31 22-06-11
Less functionality, twice the number of lines... hard pass

File: Attributes.cs (hunk 1 above)
Comment (ferib) [reply Attributes.cs]: Channel usage is only restricted when a command is tagged with this attribute.
Currently, none are tagged with it.

**Pull request closed without merging**